cloudwatch-logs-to-logstash/lambda/index.js

```diff
@@ -0,0 +1,41 @@
+const Logstash = require('./Logstash');
+const zlib = require('zlib');
+
+exports.handler = function(event, context) {
+    const payload = new Buffer(event.awslogs.data, 'base64');
+
+    zlib.gunzip(payload, function(e, result) {
+        if (e) {
+            context.fail(e);
+        } else {
+            result = JSON.parse(result.toString('utf8'));
+
+            const logs = result.logEvents;
+            const records = [];
+
+            logs.forEach(function(log) {
+                const messageParts = log.message.split('\t');
+
+                if (! isNaN(Date.parse(messageParts[0]))) {
+                    const rawMessage = messageParts[messageParts.length - 1]
+                        .replace(/(')/gm, '"')
+                        .replace(/(\n)/gm, '');
+
+                    eval('const message = ' + rawMessage);
```

Comment:
> Could we JSON parse it?

Not easily. At this stage, `rawMessage` looks like:

```
'{ message: "Downloading from ingest bucket.",  state:    { Bucket: "media-service-test-s3watcheringestbucket-XXXXX",     Key: "YYYYY.jpg" } }'
```

Note the keys are not quote enclosed, so we'd need to do some funky regex magic first (I couldn't work out how). There are also (hidden) `\t` characters between key and value :cry:.

`eval` seemed like the nicest solution (albeit horrible).

Will add a note about this.